meson.build

```diff
@@ -244,7 +244,7 @@ if not asm_option.disabled()
     if cpu_family == 'x86'
         asm_is_nasm = add_languages(
             'nasm',
-            required: false,
+            required: asm_option,
```

Comment:
hmm.. won’t this make it _harder_ to figure out where the problem is or put another way: what paths forward there are for `-Dasm=enabled` setups?  
Previously warnings and errors clearly indicate NASM is only required for assembly and build failed because assembly was explicitly requested but couldn’t be set up. Now it just fails due to not finding NASM without any context and no indication `-Dasm=disabled` can work too *(as  much as assembly-les builds shouldn’t be used for production, this might be fine/preferable for a quick test of some non-perf change)*

Reply:
Hm the error message from meson indeed leaves something to be desired:

```
meson.build:243:22: ERROR: Unknown compiler(s): [['nasm'], ['yasm']]
The following exception(s) were encountered:
Running `nasm --version` gave "[Errno 2] No such file or directory: 'nasm'"
Running `yasm --version` gave "[Errno 2] No such file or directory: 'yasm'"
```

Does libass even support yasm? The version check seems to suggest no. I have added more error messages, please check.

Reply:
> Does libass even support yasm?

If it works then only by chance. Originally libass supported only yasm, because NASM was less maintained at the time, then roles reversed and we switched to only NASM in 2017. I see yasm has some activity still albeit far less than NASM, so mayhaps it’s now able to provide all features we use, but it’s entirely untested.

Reply:
The question is, do you want to support people who want to use yasm? If so, the version check should be conditional on a compiler-id check.